### PR TITLE
[admin] extract stats handlers to separate file

### PR DESCRIPTION
**Pull request by rulex123**
Signed-off-by: Erica Manno <erica.manno@gmail.com>

Description: extract stats-related handlers from `admin.h|cc` and into separate class (part of #5505 )
Risk Level: low
Testing: pre-existing tests
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by yanavlasov** [comment]
@rulex123 Please merge master to fix clang-tidy build problem.

**Comment by rulex123** [comment]
I merged master but there still seems to be a clang-tidy problem with the build pipelines.

**Comment by jmarantz** [comment]
Sorry, can you do a master merge? I was going to do a quick check to see if I patched your source/... changes without your test/... changes do they pass.

**Comment by rulex123** [comment]
@jmarantz merged master

File: source/server/http/stats_handler.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#include "envoy/server/instance.h"
+
+#include "common/stats/histogram_impl.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Server {
+
+class StatsHandlerImpl {
```
Comment (jmarantz) [inline source/server/http/stats_handler.h:19]: This is not implementing an interface, so I'd just call it StatsHandler.

**Comment by rulex123** [comment]
@jmarantz addressed your feedback

**Pull request merged**